File: composeApp/src/androidMain/kotlin/com/sixclassguys/maplecalendar/utils/MapleWorld.kt
package com.sixclassguys.maplecalendar.utils

import androidx.annotation.DrawableRes
import com.sixclassguys.maplecalendar.R

enum class MapleWorld(val worldName: String, @DrawableRes val iconRes: Int) {

    SCANIA("스카니아", R.drawable.ic_world_scania),
    BERA("베라", R.drawable.ic_world_bera),
    LUNA("루나", R.drawable.ic_world_luna),
    ZENITH("제니스", R.drawable.ic_world_zenith),
    CROA("크로아", R.drawable.ic_world_croa),
    UNION("유니온", R.drawable.ic_world_union),
    ELYSIUM("엘리시움", R.drawable.ic_world_elysium),
    ENOSIS("이노시스", R.drawable.ic_world_enosis),
    RED("레드", R.drawable.ic_world_red),
    AURORA("오로라", R.drawable.ic_world_aurora),
    ARCANE("아케인", R.drawable.ic_world_arcane),
    NOVA("노바", R.drawable.ic_world_nova),
    CHALLENGERS("챌린저스", R.drawable.ic_world_challengers),
    CHALLENGERS2("챌린저스2", R.drawable.ic_world_challengers),
    CHALLENGERS3("챌린저스3", R.drawable.ic_world_challengers),
    CHALLENGERS4("챌린저스4", R.drawable.ic_world_challengers),
    EOS("에오스", R.drawable.ic_world_eos),
    HELIOS("핼리오스", R.drawable.ic_world_helios);

    companion object {

        fun getWorld(name: String): MapleWorld? {
            return entries.find { it.worldName == name }
        }
    }
}

File: iosApp/iosApp/utils/MapleWorld.swift
import Foundation

enum MapleWorld: Int, CaseIterable {
    case scania = 0, bera, luna, zenith, croa, union, elysium, enosis, red, aurora, arcane, nova
    
    var name: String {
        switch self {
            case .scania: return "스카니아"
            case .bera: return "베라"
            case .luna: return "루나"
            case .zenith: return "제니스"
            case .croa: return "크로아"
            case .union: return "유니온"
            case .elysium: return "엘리시움"
            case .enosis: return "이노시스"
            case .red: return "레드"
            case .aurora: return "오로라"
            case .arcane: return "아케인"
            case .nova: return "노바"
        }
    }
    
    // 이름으로 enum을 찾는 스태틱 함수
    static func from(name: String) -> MapleWorld? {
        
        return MapleWorld.allCases.first { $0.name == name }
    }
}
